added a note to avoid f-strings in logging (#25081)
matplotlib/matplotlib: doc/devel/contributing.rst

   DEBUG:matplotlib.yourmodulename:Here is some information
   DEBUG:matplotlib.yourmodulename:Here is some more detailed information

Avoid using pre-computed strings (``f-strings``, ``str.format``,etc.) for logging because of security and 
performance issues, and because they interfere with style handlers. 

> e.g. `_log.error('hello %s', 'world')` versus  `_log.error('hello {}'.format('world'))`.

Reviewer: Sorry, forgot these need to be double backticks
```suggestion
e.g. ``_log.error('hello %s', 'world')`` versus  ``_log.error('hello {}'.format('world'))``.
```

   DEBUG:matplotlib.yourmodulename:Here is some information
   DEBUG:matplotlib.yourmodulename:Here is some more detailed information

**NOTE:** Please avoid using pre-computed stirngs(``f-strings``, ``str.format``,etc.) for logging. There are security & performance
issues, it interferes with style handlers. 
See `here <https://github.com/matplotlib/matplotlib/pull/25073#issuecomment-1403711269>`_
for more details.

**Correct**

> .. code-block:: none

Reviewer 2: ```suggestion
.. code-block:: python
```

and similar for the other 2?

Reviewer 2: You also need the blank line between the `::` and the start of the code .

   DEBUG:matplotlib.yourmodulename:Here is some information
   DEBUG:matplotlib.yourmodulename:Here is some more detailed information

> **NOTE:** Please avoid using pre-computed stirngs(``f-strings``, ``str.format``,etc.) for logging. There are security & performance

Reviewer 3: ```suggestion
**NOTE:** Please avoid using pre-computed strings(``f-strings``, ``str.format``,etc.) for logging. There are security and performance
```
There is also some trailing white space it seems, although not clear where.

(I've restarted the doc-build as it looked like there was an intermittent error there, so hopefully it works this time.)

> **NOTE:** Please avoid using pre-computed stirngs(``f-strings``, ``str.format``,etc.) for logging. There are security & performance

Reviewer: ```suggestion
**NOTE:** Please avoid using pre-computed strings (``f-strings``, ``str.format``, etc.) for logging. 
There are security & performance
```

> issues, it interferes with style handlers. 

Reviewer: ```suggestion
issues, and it interferes with style handlers. 
```

   DEBUG:matplotlib.yourmodulename:Here is some information
   DEBUG:matplotlib.yourmodulename:Here is some more detailed information

> **NOTE:** Please avoid using pre-computed stirngs(``f-strings``, ``str.format``,etc.) for logging.
> There are security & performance
> issues, and it interferes with style handlers. 
> See `here <https://github.com/matplotlib/matplotlib/pull/25073#issuecomment-1403711269>`_
> for more details.
> 
> **Correct**
> 
> .. code-block:: python
> 
>    _log.error('hello %s', 'world')
> 
> **Incorrect**
> 
> .. code-block:: python
> 
>    _log.error('hello {}'.format('world'))
> 
> .. code-block:: python
> 
>    s = 'world'
>    _log.error(f'hello {s}')

Reviewer: This seems super verbose

```suggestion
Avoid using pre-computed strings (``f-strings``, ``str.format``,etc.) for logging because of security and 
performance issues, and because they interfere with style handlers. 
e.g. `_log.error('hello %s', 'world')` versus  `_log.error('hello {}'.format('world'))`.
```
